diag_test.py

```
@@ -609,322 +809,74 @@ def analyze_roc(df, truth_col, score_col, method='delong', pos_label_user=None):
-    # sklearn.roc_curve returns coordinates in ascending threshold order
-    # Specificity = 1 - FPR (correct formula)
+
+    # Plotly Figure
+    fig = go.Figure()
+    
+    # ROC Curve
+    fig.add_trace(go.Scatter(
+        x=fpr, y=tpr, 
+        mode='lines', 
+        name=f'ROC (AUC = {auc_val:.3f})',
+        line=dict(color=COLORS['primary'], width=3)
```

Ruff (0.14.10)

Unnecessary `dict()` call (rewrite as a literal)

Rewrite as a literal

(C408)